controller/controller-worker/src/main/java/com/pinterest/rocksplicator/controller/tasks/RetryTask.java

```diff
@@ -93,7 +95,11 @@ public TaskBase getTask() {
       return task;
     }
 
-    public RetryTask.Param setTask(TaskBase task) {
+    public RetryTask.Param setTask(TaskBase task) throws Exception {
+      // We don't allow RetryTask wraps a RecurringTask.
+      if (TaskFactory.getWorkerTask(task) instanceof RecurTask) {
```

Comment:
nit: Or you can override the retry() method in RecurTask and throw UnsupportedException there.